types/lru-cache/index.d.ts

```diff
@@ -3,46 +3,47 @@
 // Definitions by: Bart van der Schoor <https://github.com/Bartvds>
 //                 BendingBender <https://github.com/BendingBender>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.3
 
 declare class LRUCache<K, V> implements Iterable<[K, V]> {
-    constructor(options?: LRUCache.Options<K, V>);
+    constructor(options: LRUCache.Options<K, V>);
 
     /**
      * Return total length of objects in cache taking into account `length` options function.
      *
-     * @deprecated use `cache.size` instead
-     * @since 7.0.0
+     * @deprecated since 7.0 use `cache.size` instead
      */
-    readonly length: number;
+    public readonly length: number;
 
     // values populated from the constructor options
-    readonly max: number;
-    readonly maxSize: number;
-    readonly sizeCalculation: LRUCache.SizeCalculator<K, V> | undefined;
-    readonly dispose: LRUCache.Disposer<K, V>;
-    readonly disposeAfter: LRUCache.Disposer<K, V> | null;
-    readonly noDisposeOnSet: boolean;
-    readonly ttl: number;
-    readonly ttlResolution: number;
-    readonly ttlAutopurge: boolean;
-    readonly allowStale: boolean;
-    readonly updateAgeOnGet: boolean;
+    public readonly max: number;
+    public readonly maxSize: number;
+    public readonly sizeCalculation: LRUCache.SizeCalculator<K, V> | undefined;
+    public readonly dispose: LRUCache.Disposer<K, V>;
+    /**
+     * @since 7.4.0
+     */
+    public readonly disposeAfter: LRUCache.Disposer<K, V> | null;
+    public readonly noDisposeOnSet: boolean;
+    public readonly ttl: number;
+    public readonly ttlResolution: number;
+    public readonly ttlAutopurge: boolean;
+    public readonly allowStale: boolean;
+    public readonly updateAgeOnGet: boolean;
 
```

Comment:
```suggestion
    /**
     * since: 7.6.0
     */
     public readonly fetchMethod: LRUCache.Fetcher<K, V> | null;
```

Then later on somewhere, I think this would define it:

```typescript
    type Fetcher<K, V> = (key: K, staleKey: K | undefined) => Promise<V>;
```